### PR TITLE
solution was added

### PR DESCRIPTION
**Pull request by Maksym845**
- [DEMO LINK](https://Maksym845.github.io/react_keyboard/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+
+export class App extends Component<{}, State> {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => (
+      this.handler(event)));
+  }
```
Comment (Yeliseiev) [inline src/App.tsx:15]:
```suggestion
  componentDidMount() {
    document.addEventListener('keyup', this.handler);
  }
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => (
+      this.handler(event)));
+  }
+
+  handler = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+  };
+
+  render(): ReactNode {
```
Comment (Yeliseiev) [inline src/App.tsx:26]: Why do you need 'ReactNode' here?

**Review by Yeliseiev — APPROVED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,37 @@
...
-  </div>
-);
+type State = {
+  pressedKey: string | null;
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    pressedKey: null,
+  };
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:10]: or you could set empty string here

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,37 @@
...
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => (
+      this.handler(event)));
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:14]: just pass handler here, no need to create extra function

File: src/App.tsx [hunk 5]
```diff
@@ -1,7 +1,37 @@
...
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => (
+      this.handler(event)));
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:19]: same here

File: src/App.tsx [hunk 6]
```diff
@@ -1,7 +1,37 @@
...
+
+  handler = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+  };
+
+  render(): ReactNode {
+    return (
+      <div className="App">
+        <div className="App__message">
+          {this.state.pressedKey !== null
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:30]: use destructuring assignment for state properties

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**